### PR TITLE
Add pytest-examples and make examples in docs testable

### PR DESCRIPTION
**Pull request by hramezani**
Fixes https://github.com/pydantic/pydantic-settings/issues/33

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -33,12 +34,12 @@ class SubModel(BaseModel):
 
 
 class Settings(BaseSettings):
-    auth_key: str = Field(validation_alias='my_auth_key')
-    api_key: str = Field(validation_alias='my_api_key')
+    auth_key: str = Field('', validation_alias='my_auth_key')
+    api_key: str = Field('', validation_alias='my_api_key')
```
Comment (samuelcolvin) [inline docs/index.md:38]: I think the idea here was that these fields were required - then set via env vars. Probably best if you can set the appropriate env vars in the test so this works without modifiation.

File: docs/index.md [hunk 2]
```diff
@@ -53,13 +54,14 @@ class Settings(BaseSettings):
     # export my_prefix_more_settings='{"foo": "x", "apple": 1}'
     more_settings: SubModel = SubModel()
 
-    model_config = ConfigDict(env_prefix = 'my_prefix_')  # defaults to no prefix, i.e. ""
+    model_config = ConfigDict(env_prefix='my_prefix_')  # defaults to no prefix, i.e. ""
 
-print(Settings().model_dump())
+
+pprint(Settings().model_dump())
```
Comment (samuelcolvin) [inline docs/index.md:60]: i think you should just use `print`, pytest-examples should take care of formatting the output with `black`.

**Review by samuelcolvin — CHANGES_REQUESTED** [review]
I don't see the test code using pytest-examples - I guess you need to add that file?

File: docs/index.md [hunk 3]
```diff
@@ -165,7 +172,14 @@ class Settings(BaseSettings):
     model_config = ConfigDict(env_nested_delimiter='__')
 
 
-print(Settings().model_dump())
+# Set environment variables
+os.environ['V0'] = '0'
+os.environ['SUB_MODEL'] = '{"v1": "json-1", "v2": "json-2"}'
+os.environ['SUB_MODEL__V2'] = 'nested-2'
+os.environ['SUB_MODEL__V3'] = '3'
+os.environ['SUB_MODEL__DEEP__V4'] = 'v4'
```
[the remark this answers is not in the document]
Comment (hramezani) [reply docs/index.md:180]: Sorry, didn't get this comment

File: docs/index.md (hunk 2 above)
Comment (hramezani) [reply docs/index.md:60]: Didn't work with `print`.

```
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_docs_examples[docs/index.md:14-72] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Print output changed code:
  --- before
  +++ after
  @@ -56,30 +56,21 @@
       model_config = ConfigDict(env_prefix='my_prefix_')  # defaults to no prefix, i.e. ""
 
 
   print(Settings().model_dump())
   """
  -{
  -    'auth_key': '',
  -    'api_key': '',
  -    'redis_dsn': Url('redis://user:pass@localhost:6379/1'),
  -    'pg_dsn': Url('postgres://user:pass@localhost:5432/foobar'),
  -    'amqp_dsn': Url('amqp://user:pass@localhost:5672/'),
  -    'special_function': <built-in function cos>,
  -    'domains': set(),
  -    'more_settings': {'foo': 'bar', 'apple': 1},
  -}
  +{'auth_key': '', 'api_key': '', 'redis_dsn': Url('redis://user:pass@localhost:6379/1'), 'pg_dsn': Url('postgres://user:pass@localhost:5432/foobar'), 'amqp_dsn': Url('amqp://user:pass@localhost:5672/'), 'special_function': <built-in function cos>, 'domains': set(), 'more_settings': {'foo': 'bar', 'apple': 1}}
   """
```

Comment (hramezani) [reply docs/index.md]: Ok, I've replaced it with `print` but it doesn't work with formatted output.
we can ignore it for now.

File: docs/index.md (hunk 1 above)
Comment (hramezani) [reply docs/index.md]: Done

File: docs/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (samuelcolvin) [reply docs/index.md:188]: e.g. set a bunch of env variables before running all or specific tests, then remove them after.

File: pyproject.toml [hunk 4]
```diff
@@ -86,21 +79,20 @@ source = [
     'pydantic_settings/',
 ]
 
+[tool.ruff]
+line-length = 120
+extend-select = ['Q', 'RUF100', 'C90', 'UP', 'I']
+flake8-quotes = {inline-quotes = 'single', multiline-quotes = 'double'}
+mccabe = { max-complexity = 14 }
+isort = { known-first-party = ['pydantic_settings', 'tests'] }
+target-version = "py37"
```
Comment (samuelcolvin) [inline pyproject.toml:88]:
```suggestion
target-version = 'py37'
```

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM, I committed to fix the issue with formatting the first example.

File: docs/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hramezani) [reply docs/index.md]: I've added a fixture

**Comment by hramezani** [comment]
please review

**Comment by hramezani** [comment]
@samuelcolvin It's ready

**Comment by samuelcolvin** [comment]
Thanks so much.

**Pull request merged**